test/metabase/related_test.clj

```diff
@@ -0,0 +1,75 @@
+(ns metabase.related-test
+  (:require [expectations :refer :all]
+            [metabase.api.common :as api]
+            [metabase.models
+             [card :refer [Card]]
+             [database :refer [Database]]
+             [table :refer [Table]]]
+            [metabase.related :refer :all :as r]
+            [metabase.test.util :as tu]
+            [metabase.test.data.users :as users]
+            [toucan.util.test :as tt]))
+
+(expect
+  #{[:field-id 1] [:metric 1] ["FIELD-ID" 2] ["segment" 1]}
+  (#'r/collect-context-bearing-forms [[:> [:field-id 1] 3]
+                                      ["and" [:= ["FIELD-ID" 2] 2]
+                                       ["segment" 1]]
+                                      [:metric 1]]))
+
+(expect
+  [0.5
+   0.0
+   1.0]
+  (tt/with-temp* [Card [{card-id-1 :id}
+                        {:name          (tu/random-name)
+                         :dataset_query {:query {:aggregation [:sum [:field-id 1]]
+                                                 :breakout [[:field-id 2]]}}}]
+                  Card [{card-id-2 :id}
+                        {:name          (tu/random-name)
+                         :dataset_query {:query {:aggregation [:sum [:field-id 3]]
+                                                 :breakout [[:field-id 2]]}}}]
+                  Card [{card-id-3 :id}
+                        {:name          (tu/random-name)
+                         :dataset_query {:query {:aggregation [:sum [:field-id 3]]
+                                                 :breakout [[:field-id 4]]}}}]]
+    (map double [(#'r/similarity (Card card-id-1) (Card card-id-2))
+                 (#'r/similarity (Card card-id-1) (Card card-id-3))
+                 (#'r/similarity (Card card-id-1) (Card card-id-1))])))
+
+(expect
+  1
+  (binding [api/*current-user-id* (users/user->id :crowberto)]
```

Comment:
can you add comments explaining what exactly is being tested here? Otherwise it's hard to understand what's going on

Reply:
Added (and slightly reworked the test).